c-dependencies/js-compute-runtime/builtins/text-decoder.cpp

```diff
@@ -0,0 +1,64 @@
+#include "builtins/text-decoder.h"
+#include "js-compute-builtins.h"
+
+namespace builtins {
+
+bool TextDecoder::decode(JSContext *cx, unsigned argc, JS::Value *vp) {
+  METHOD_HEADER(1)
+
+  // Default to empty string if no input is given.
+  if (args[0].isUndefined()) {
+    args.rval().set(JS_GetEmptyStringValue(cx));
+    return true;
+  }
+
+  auto data = value_to_buffer(cx, args[0], "TextDecoder#decode: input");
+  if (!data.has_value()) {
+    return false;
+  }
+
+  JS::RootedString str(
+      cx, JS_NewStringCopyUTF8N(cx, JS::UTF8Chars((char *)data->data(), data->size())));
+  if (!str)
+    return false;
+
+  args.rval().setString(str);
+  return true;
+}
+
+bool TextDecoder::encoding_get(JSContext *cx, unsigned argc, JS::Value *vp) {
+  METHOD_HEADER(0)
+
+  JS::RootedString str(cx, JS_NewStringCopyN(cx, "utf-8", 5));
```

Comment:
non-blocking: Same here as in text-encoder.cpp, this could be a persistent string instead of a newly created string on each call

Reply:
I think that's a good change, but let's do it in a follow :+1: